### ISSUE TITLE
Allow the loop to end early?

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/p-times#1]
Normal for loops can end early by using `break` or `return`. Might be useful to have the ability here too. Does that make sense?
## 

Possible implementations:
### 1

Add a method for ending the iteration in the second argument to `mapper`: `mapper(index, stop)`:

``` js
pTimes(5, (i, stop) => {
    if (shouldStop(i)) {
        stop();
        return;
    }

    return `foo ${i}`;
}).then();
```
### 2

Add ability to throw a special error to end the iteration without rejection the promise:

``` js
pTimes(5, i => {
    if (shouldStop(i)) {
        throw new pTimes.StopError();
    }

    return `foo ${i}`;
}).then();
```
### 3

Add ability to return special `Symbol` to end the iteration:

``` js
pTimes(5, i => {
    if (shouldStop(i)) {
        return pTimes.StopSymbol;
    }

    return `foo ${i}`;
}).then();
```
## 

While 2 and 3 are more verbose, they have the benefit of working even in a nested promise chain.

**Comment by fregante** [comment]
``` js
return pTimes.StopSymbol;
```

is not substantially more verbose than:

``` js
(i, stop) => {
        stop();
        return;
```

Plus it has the advantage of being global so you don't have to pass a `stop` function around. 

Alternatively:

``` js
return pTimes['⛔️']; // 😜
```

**Comment by sindresorhus** [comment]
Thanks for the feedback. I'll go with returning a Symbol and `pTimes.end` as the name.

Pull request welcome from anyone :)

**Comment by Richienb** [comment]
Because this module wraps `p-map`, this issue should be moved there.

**Comment by sindresorhus** [comment]
Yeah, we need to resolve https://github.com/sindresorhus/p-map/issues/31 first.
